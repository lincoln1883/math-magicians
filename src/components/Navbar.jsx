import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1, class: 'nav-item', path: '/', text: 'Home', symbol: ' |',
    },
    {
      id: 2, class: 'nav-item', path: '/calculator', text: 'Calculator', symbol: ' |',
    },
    {
      id: 3, class: 'nav-item', path: '/quotes', text: 'Quotes',
    },
  ];
  return (
    <nav className="nav-bar">
      <h1 className="nav-title">Math Magicians</h1>
      <ul className="nav-list">
        {links.map((link) => (
          <li key={link.id}>
            <NavLink className={link.class} to={link.path}>
              {link.text}
              {link.symbol}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
