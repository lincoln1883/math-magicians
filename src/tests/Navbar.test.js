import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  it('renders Navbar component', () => {
    const tree = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders Navbar component with props', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const element = screen.getByText('Math Magicians');
    expect(element).toMatchSnapshot();
  });
});
