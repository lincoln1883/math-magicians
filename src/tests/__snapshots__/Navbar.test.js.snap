// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav-bar"
      >
        <h1
          class="nav-title"
        >
          Math Magicians
        </h1>
        <ul
          class="nav-list"
        >
          <li>
            <a
              aria-current="page"
              class="nav-item active"
              href="/"
            >
              Home
               |
            </a>
          </li>
          <li>
            <a
              class="nav-item"
              href="/calculator"
            >
              Calculator
               |
            </a>
          </li>
          <li>
            <a
              class="nav-item"
              href="/quotes"
            >
              Quotes
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav-bar"
    >
      <h1
        class="nav-title"
      >
        Math Magicians
      </h1>
      <ul
        class="nav-list"
      >
        <li>
          <a
            aria-current="page"
            class="nav-item active"
            href="/"
          >
            Home
             |
          </a>
        </li>
        <li>
          <a
            class="nav-item"
            href="/calculator"
          >
            Calculator
             |
          </a>
        </li>
        <li>
          <a
            class="nav-item"
            href="/quotes"
          >
            Quotes
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Navbar renders Navbar component with props 1`] = `
<h1
  class="nav-title"
>
  Math Magicians
</h1>
`;
